java/java.graph/src/org/netbeans/modules/java/graph/FruchtermanReingoldLayout.java

```diff
@@ -150,7 +150,7 @@ private void init() {
         bounds = new Rectangle(magicSizeConstant  + (magicSizeMultiplier * nds), 
                                magicSizeConstant  + (magicSizeMultiplier * nds)); //g.getMaximumBounds();
         temp = bounds.getWidth() / 10;
-        forceConstant = 0.75 * Math.sqrt(bounds.getHeight() * bounds.getWidth() / nds);
+        forceConstant = 1.75 * Math.sqrt(bounds.getHeight() * bounds.getWidth() / nds);
```

Comment:
not sure about changing the weights, but the graph looks somewhat similar still so its probably fine

Reply:
It seems to make little different with a smaller graph, but with the larger projects I tested (that had the infinite loop before) it's much better and makes them somewhat useful.  I suspect it's mitigating some of the problems with the reduced search for free space, but as that never completes to compare it with, it's a little hard to know for sure.

Reply:
yeah there is a lot which could be done UX wise. Drag on empty space should move the viewport like everyone will intuitively try doing after using maps for so long. I think scroll wheel should probably zoom by default without having to hold ctrl (scrollbars are off anyway, which would be fine if drag would work), etc.

i would like to have the output of
```
 mvn eu.maveniverse.maven.plugins:toolbox:tree -DverboseTree=true 

```
and
```
mvn eu.maveniverse.maven.plugins:toolbox:plugin-tree -DverboseTree=true
```
(possibly more) in swing trees. with colors, filters etc. Since i don't find graphs very useful for purposes like this.

Reply:
I'll try and have a look at those in context of next release. Have used the visual library a lot. Obviously not for delivery though. 

Drag behaviour is there, just using mouse button 2. Default makes sense if the rectangular select action is used, but it isn't here. Will look at changing.

May also consider adding the embedded satellite view you can see in front page image on https://www.praxislive.org/